master_mind.rb

```diff
@@ -1,37 +1,57 @@
-@cor =  [1,2,3,4,5,6]
-@cor.shuffle!
+@cor = [1, 2, 3, 4, 5, 6].shuffle!
+puts @cor
 
 class Master_mind
-  
+  def initialize
+    @max_rodadas = 0
+    @tentativas = 0
+  end
+
   def verifica_cor_posicao(cor,cor2)
     cont =0
     cont2 =0
     cor.each_with_index do |valor, index|
-      cor2.each_with_index do |cor, i| 
+      cor2.each_with_index do |cor, i|
```

Comment:
esse loop aqui tem que ser feito com if e else pra evitar  recontagem de sequência, eu não rodei o código mas eu recomendaria tu rodar um caso em que os dois arreis são iguais menos o ultimo, tipo senha é [1,1,1,2] e o jogador inseriu [1,1,1,3]. Parece que vai retornar um valor bem maior do que deveria pelo o fluxo que eu to vendo

então apenas sugestão de dar uma olhada nesse loop